### PR TITLE
Refactor tests to run on persistant databases.

### PR DESCRIPTION
**Pull request by nrmay**
These changes fix the following issues;
1. Remove hard coded pk ids from assert statements. The starting pk id on the auto increment field is not reset between tests. 
2. Added sorting of result sets to tardis/tardis_portal/tests/test_views.py to prevent tests failing because the database returns ids in a non-deterministic method.
3. Fixed two bugs in the change_user_permissions() method of tardis/tardis_portal/views/authorisation.py. 
    3.1. The change was always rejected if the experiment only had one owner, even if the owner was not the user who's permissions were being changed.
   3.2. The submitting user's permissions were always being updated, even if the request was for a different username.
   3.3. Added testChangeUserPermissions() to tardis/tardis_portal/tests/test_authorisation.py
4. Added test settings files for Postgresql and MySQL.
5. Various pylint errors.

Test and pylint results can be viewed on [semaphoreci.com/nrmay/mytardis-on-postgresql](https://semaphoreci.com/nrmay/mytardis-on-postgresql)

### CODE REVIEW

File: tardis/apps/oaipmh/tests/provider/test_experiment.py [hunk 1]
```diff
@@ -229,18 +237,21 @@ def testListRecords(self):
...
-                                        'id': 1, \
+                                        'id': ps_id, \
                                         'title': 'Google'}])
                 else:
                     expect(metadata.getField('related_info')).to_equal([{}])
             else:
                 expect(header.identifier()).to_contain(str(self._user.id))
                 expect(header.datestamp().replace(tzinfo=pytz.utc))\
-                    .to_equal(get_local_time(self._user.last_login))
+                    .to_estRecordsqual(get_local_time(self._user.last_login))
```
Comment (grischa) [inline tardis/apps/oaipmh/tests/provider/test_experiment.py]: Is this a typo?

File: tardis/apps/push_to/views.py [hunk 2]
```diff
@@ -1,4 +1,5 @@
 import json
+import requests
 
 import requests
```
Comment (grischa) [inline tardis/apps/push_to/views.py]: Duplicate import

**Comment by grischa** [comment]
Thanks, looks good. I'll merge once you fix the typo and duplicate import.

**Comment by nrmay** [comment]
Fixes done and tests passing.

**Pull request merged**